force-app/main/default/classes/ContactRollupsSchedulable_TEST.cls

```diff
@@ -9,54 +9,77 @@
 
 @isTest
 private class ContactRollupsSchedulable_TEST {
-    private static BasicStub attendanceServiceStub = new BasicStub(
-        ServiceDeliveryService.class
-    );
-
     @testSetup
     static void setup() {
         TestUtil.turnOffFeatureGates();
         TestDataFactory.generateAttendanceData('Monthly');
     }
 
-    // @IsTest
-    // private static void shouldCallServiceWithContacts() {
-    //     List<String> attendanceStatuses = new ServiceDeliveryService().attendanceStatuses;
-
-    //     List<Contact> contactsWithServiceDeliveries = [
-    //         SELECT
-    //             Id,
-    //             (
-    //                 SELECT Contact__c, AttendanceStatus__c, DeliveryDate__c, Quantity__c
-    //                 FROM Service_Deliveries__r
-    //                 WHERE
-    //                     AttendanceStatus__c IN :attendanceStatuses
-    //                     OR (DeliveryDate__c <= TODAY
-    //                     AND Quantity__c > 0)
-    //             )
-    //         FROM Contact
-    //     ];
-    //     System.assert(
-    //         !contactsWithServiceDeliveries.isEmpty(),
-    //         'Expected test data to be queried.'
-    //     );
-
-    //     Test.startTest();
-
-    //     ContactRollupsSchedulable rollups = new ContactRollupsSchedulable();
-    //     attendanceServiceStub.withReturnValue(
-    //         'getAttendanceStatuses',
-    //         attendanceStatuses
-    //     );
-    //     ContactRollupsSchedulable.deliveryService = (ServiceDeliveryService) attendanceServiceStub.createMock();
-    //     rollups.execute(null);
-
-    //     Test.stopTest();
-
-    //     attendanceServiceStub.assertCalledWith(
-    //         'doRollups',
-    //         new List<Type>{ SObjectType.class, List<SObject>.class },
-    //         new List<Object>{ Contact.SObjectType, contactsWithServiceDeliveries }
-    //     );
-    // }
+    @IsTest
+    private static void shouldQueryContactsAndExecuteBatch() {
+        List<Contact> contactsWithRollupData = [
+            SELECT Id
+            FROM Contact
+            WHERE
+                NumPresentServiceDeliveries__c >= 0
+                OR NumAbsentServiceDeliveries__c >= 0
+                OR LastServiceDate__c != NULL
+                OR ConsecutiveAbsences__c >= 0
+        ];
+        System.assert(
+            !contactsWithRollupData.isEmpty(),
+            'Expected test data to be queried.'
+        );
+
+        Test.startTest();
+
+        ContactRollupsSchedulable contactRollups = new ContactRollupsSchedulable();
+        Database.executeBatch(contactRollups);
```

Comment:
@bmuniswamy - It isn't clear what we are testing here. The assert is only a sanity check that we have records. The test itself is manually scheduling the batch, is there an outcome we expect?

Reply:
@jjbennett, The outcome would be that there are rollups on the contact. I thought since we will be testing that in the other inner classes left the sanity check in there.

Reply:
Currently we are just running the batch but we should also assert the expected outcome.

Reply:
@bmuniswamy updates looking great to these schedulable tests. Thanks again!